Source/Dafny/AST/DafnyAst.cs

```diff
@@ -13,14 +13,60 @@
 using System.Diagnostics.Contracts;
 using System.Numerics;
 using System.Linq;
-using Microsoft.Boogie;
 using System.Diagnostics;
 using System.Threading;
+using Microsoft.Boogie;
 
 namespace Microsoft.Dafny {
   [System.AttributeUsage(System.AttributeTargets.Field)]
   public class FilledInDuringResolutionAttribute : System.Attribute { }
 
+  public interface IToken : Microsoft.Boogie.IToken {
+    // Inherited for now.
+    /*
+    int kind { get; set; }
+    string filename { get; set; }
+    int pos { get; set; }
+    int col { get; set; }
+    int line { get; set; }
+    string val { get; set; }
+    bool IsValid { get; }*/
+    string leadingTrivia { get; set; }
+    string trailingTrivia { get; set; }
+  }
+
+  public record Token : IToken {
+    public Token peekedTokens; // Used only internally when the scanner "peeks" tokens. Normallly null at the end of parsing
+    public static readonly IToken NoToken = (IToken)new Token();
+
+    public Token() => this.val = "anything so that it is nonnull";
```

Comment:
Why do we need this constructor?

Reply:
You could use the constructor that takes parameters and pass default values, or at least make that constructor private.

Reply:
My bad, it is also used 11 times in generated code (5 Parser.cs and 6 Parser.cs), 1 time in the translator.
Do you agree it's not appropriate to do this change then?